src/components/ContactMe/ContactInfoCard/index.jsx

```diff
@@ -1,20 +1,45 @@
-import React from "react";
+import React, { useState, useEffect } from "react";
 import "./styles.css";
 import Text from "../../Text";
 import Icon from "../../Icon";
 
-const ContactInfoCard = ({ text, name }) => {
+const ContactInfoCard = ({ text, name, link, linkLabel, showCopy }) => {
+  const [isCopied, setIsCopied] = useState(false);
+  useEffect(() => {
+    const timer = setTimeout(() => setIsCopied(false), 2000); // Hide message after 2 seconds
+    return () => clearTimeout(timer);
+  }, [isCopied]);
+  const handleCopy = () => {
+    navigator.clipboard
+      .writeText("edgar.itmember@gmail.com")
+      .then(() => {
+        setIsCopied(true);
```

Comment:
you need to call setTimeout here, but then clearTimeout inside the useEffect. that will be the correct solution of the case.